R/check_gene_expression.R

```diff
@@ -0,0 +1,111 @@
+
+#' @title Check Gene Expression.
+#'
+#' @description This function determines which samples have expression data available in the merge and drop redundant data while consistently prioritizing by protocol and nucleic acid source.
+#' 
+#' 
+#' @param these_samples_metadata The data frame with sample metadata. Usually output of the get_gambl_metadata().
+#' @param verbose Set to TRUE mainly for debugging
+#' 
+#' @return A data frame with a row for each non-redundant RNA-seq result and the following columns:
+#' 
+#' \describe{
+#'   \item{mrna_sample_id}{The unique sample_id value that will match a single row from the GAMBL metadata where seq_type is mrna. }
+#'   \item{biopsy_id}{The unique identifier for the source of nucleic acids.}
+#'   \item{sample_id}{Identical to mrna_sample_id}
+#'   \item{capture_sample_id}{When this biopsy has capture/exome data in the GAMBL metadata, the value will be the sample_id for that data. NA otherwise.}
+#'   \item{genome_sample_id}{When this biopsy has genome data in the GAMBL metadata, the value will be the sample_id for that data. NA otherwise.}
+#'   \item{patient_id}{The anonymized unique identifier for this patient. For BC samples, this will be Res ID.}
+#'   \item{seq_type}{The assay type used to produce this data (will always be "mrna" in this case)}
+#'   \item{protocol}{Specifies the RNA-seq library construction protocol.}
+#'   \item{ffpe_or_frozen}{Specifies the way the source of nucleic acids was preserved. Either FFPE or frozen.}}
+#' 
+check_gene_expression = function(verbose=F,show_linkages=F){
+  # We start with the minimal metadata that corresponds to the tidy expression file (sample_metadata.tsv)
+  metadata_file = check_config_value(config::get("results_merged")$tidy_expression_metadata)
+  metadata_file = paste0(check_config_value(config::get("project_base")),metadata_file)
+  expression_data_rows = suppressMessages(read_tsv(metadata_file)) %>% 
+                                            dplyr::select(sample_id,tissue_status,seq_type,patient_id,biopsy_id,protocol,ffpe_or_frozen,cohort) %>%
+                                            dplyr::rename(c("mrna_sample_id"="sample_id"))
+  
+
+  
+  collapse_duplicates = TRUE
+
+  if(verbose){
+    nrows = nrow(expression_data_rows)
+    print(paste("starting with",nrows,"rows"))
+  }
+
+  
+  mark_duplicates_biopsy = function(ew){
```

Comment:
This is fine and it works, but it'll never not bug me that it could be accomplished with 
```
expression_data_rows %>% 
    group_by(patient_id, biopsy_id) %>% 
    mutate(multi_exp = n() > 1)) %>% # Returns TRUE/FALSE but could be mutated to 1/0 if really needed
    ungroup()
````